commands/files/serialfile.go

```diff
@@ -84,10 +84,25 @@ func (f *serialFile) NextFile() (File, error) {
 	// open the next file
 	fileName := fp.Join(f.name, stat.Name())
 	filePath := fp.Join(f.path, stat.Name())
+	st, err := os.Lstat(filePath)
```

Comment:
the recursive flag is special cased, which makes adding the dereference flag really annoying. I think we should push this PR without the flag, and pick a default for what to do in the case of 'ipfs add thisIsASymlink'